extensions/swagger-ui/deployment/src/main/java/io/quarkus/swaggerui/deployment/SwaggerUiProcessor.java

```diff
@@ -139,5 +313,236 @@ public void registerSwaggerUiServletExtension(SwaggerUiRecorder recorder,
          */
         @ConfigItem(defaultValue = "true")
         boolean enable;
+
+        /**
+         * The urls that will be included as options. By default the OpenAPI path will be used.
+         * Here you can override that and supply multiple urls that will appear in the TopBar plugin.
+         * Format is name=url,name=url (if name is omitted, url will be use as name)
+         */
+        @ConfigItem
+        Optional<String[]> urls;
+
+        /**
+         * If urls option is used, this will be the name of the default selection.
+         */
+        @ConfigItem(name = "urlsPrimaryName")
+        Optional<String> urlsPrimaryName;
+
+        /**
+         * The html title for the page.
+         */
+        @ConfigItem
+        Optional<String> title;
+
+        /**
+         * Swagger UI theme to be used.
+         */
+        @ConfigItem
+        Optional<ThemeHref> theme;
+
+        /**
+         * A footer for the html page. Nothing by default.
+         */
+        @ConfigItem
+        Optional<String> footer;
+
+        /**
+         * If set to true, enables deep linking for tags and operations.
+         */
+        @ConfigItem(name = "deepLinking")
+        Optional<Boolean> deepLinking;
+
+        /**
+         * Controls the display of operationId in operations list. The default is false.
```

Comment:
```suggestion
         * Controls the display of operationId in operations list.
```

Reply:
All config options is made Optional, because what we are doing is building the html. Even though some options, like displayOperationId, has a default, the default is in the swagger-ui code, so we just want to change the html, if set. So the default for us is nothing (not in the html) and when set, we add the value to the html. Else, even with the default case, we are parsing variables that is not needed when we create the html. Hope that makes sense.

Reply:
OK let's keep it this way then.

Reply:
While I understand this approach, what I don't like is that the default values are not properly documented.

But let's try this way and we can still revisit later if people complain.

Reply:
Cool. Resolving here then. Thanks :)